src/main/java/commission/service/PaymentCalculatorService.java

```diff
@@ -0,0 +1,61 @@
+package commission.service;
+
+import commission.dao.CrudDao;
+import commission.entity.Commission;
+import commission.entity.Sale;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class PaymentCalculatorService {
+
+    private  final CrudDao<Sale> saleDao;
+    private final CrudDao<Commission> commissionDao;
+
+    public PaymentCalculatorService(CrudDao<Sale> saleDao, CrudDao<Commission> commissionDao) {
+        this.saleDao = saleDao;
+        this.commissionDao = commissionDao;
+    }
+
+    public void completeCommission(Commission commission){
+        Optional<Sale> sale = this.saleDao.findById(commission.salesId());
```

Comment:
It doesn't look like the `sale` variable is used in this method?

Reply:
It is not, as it seems, I removed it.